src/coreclr/jit/valuenum.cpp

```diff
@@ -4285,10 +4285,20 @@ void ValueNumStore::GetConstantBoundInfo(ValueNum vn, ConstantBoundInfo* info)
     }
 }
 
+//------------------------------------------------------------------------
+// IsVNPositiveInt32Constant: returns true iff vn is a known Int32 constant that is greater then 0
+//
+// Arguments:
+//    vn - Value number to query
+bool ValueNumStore::IsVNPositiveInt32Constant(ValueNum vn)
+{
+    return IsVNInt32Constant(vn) && (ConstantValue<INT32>(vn) > 0);
+}
+
 //------------------------------------------------------------------------
 // IsVNArrLenUnsignedBound: Checks if the specified vn represents an expression
-//    such as "(uint)i < (uint)len" that implies that the index is valid
-//    (0 <= i && i < a.len).
+//    such as "(uint)i < (uint)len", constant < (uint)len or "constant <= len" that
```

Comment:
Nit: I would spend more lines on it to make it clearer, like:
```
// IsVNArrLenUnsignedBound: Checks if the specified vn represents an expression
//   of one of the following form:
//     - `(uint)i < (uint)len` that implies `(0 <= i <= len)`,
//      - `const < (uint)len` that implies `(len > const)`,
//      - `const <= len` that implies `(len > const - 1)`.
```
or fix the existing inconsistency, like "" used for the "(uint)i < (uint)len", "constant <= len", but not for constant < (uint)len; 
>> (0 <= i && i < a.len)

where does `a. ` come from?